examples/print_platform/main.cpp

```diff
@@ -0,0 +1,485 @@
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_MAC ";
+#if defined(PLATFORM_MAC)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_EMSCRIPTEN ";
+#if defined(PLATFORM_EMSCRIPTEN)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_CLANG ";
+#if defined(PLATFORM_CLANG)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_GCC_COMPATIBLE ";
+#if defined(PLATFORM_GCC_COMPATIBLE)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_X86 ";
+#if defined(PLATFORM_X86)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_CLANG_X86 ";
+#if defined(PLATFORM_CLANG_X86)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_GCC_COMPATIBLE_X86 ";
+#if defined(PLATFORM_GCC_COMPATIBLE_X86)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_ARM ";
+#if defined(PLATFORM_ARM)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_CLANG_ARM ";
+#if defined(PLATFORM_CLANG_ARM)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_GCC_COMPATIBLE_ARM ";
+#if defined(PLATFORM_GCC_COMPATIBLE_ARM)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_MIPS ";
+#if defined(PLATFORM_MIPS)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_CLANG_MIPS ";
+#if defined(PLATFORM_CLANG_MIPS)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_GCC_COMPATIBLE_MIPS ";
+#if defined(PLATFORM_GCC_COMPATIBLE_MIPS)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_ASMJS ";
+#if defined(PLATFORM_ASMJS)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_CLANG_ASMJS ";
+#if defined(PLATFORM_CLANG_ASMJS)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_GCC_COMPATIBLE_ASMJS ";
+#if defined(PLATFORM_GCC_COMPATIBLE_ASMJS)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_GCC ";
+#if defined(PLATFORM_GCC)
+    std::cout << "1";
+#else
+    std::cout << "0";
+#endif
+    std::cout << std::endl;
+
+    std::cout << "PLATFORM_GCC_COMPATIBLE ";
```

Comment:
This section is duplicated, and I see more of those. Can you remove these duplicates to make the output cleaner?